packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -24,12 +25,14 @@ pub struct Abigen {
     /// Generate no-std safe code
     no_std: bool,
 
-    /// The contract name as an identifier.
-    contract_name: Ident,
+    /// The contract or script name as an identifier.
+    program_name: Ident,
 
     abi: ProgramABI,
 
     types: HashMap<usize, TypeDeclaration>,
+
+    is_script: bool,
```

Comment:
I would split `functions()` into two methods, and avoid having unnecessary indirection. Noting it down on this PR before I forget.

Reply:
I'm not sure what you mean

Reply:
Instead of having the method `functions()` exhibit two behaviours depending on this field, we split it into two methods. `expand_contract` and `expand_script` would then call the corresponding method. 
Ideally, we remove this field entirely.